packages/admin/admin-date-time/src/DateTimePicker.tsx

```diff
@@ -0,0 +1,145 @@
+import { Calendar, Time } from "@comet/admin-icons";
+import { ComponentsOverrides, FormControl, InputAdornment, Theme } from "@mui/material";
+import { createStyles, WithStyles, withStyles } from "@mui/styles";
+import * as React from "react";
+
+import { DatePicker, DatePickerProps } from "./DatePicker";
+import { getDateValueWithTime, getTimeStringFromDate } from "./helpers/timePickerHelpers";
+import { TimePicker, TimePickerProps } from "./TimePicker";
+
+export interface DateTimePickerComponentsProps {
+    datePicker?: Omit<DatePickerProps, "onChange" | "onBlur" | "value" | "className" | "inputRef" | "fullWidth">;
+    timePicker?: Omit<TimePickerProps, "onChange" | "onBlur" | "value" | "className" | "inputRef" | "fullWidth">;
+}
+
+export interface DateTimePickerProps {
+    onChange?: (value?: Date) => void;
+    value?: Date;
+    clearable?: boolean;
+    componentsProps?: DateTimePickerComponentsProps;
+}
+
+function DateTimePicker({
+    classes,
+    onChange,
+    value,
+    componentsProps = {},
+    ...propsForBothTimePickers
+}: DateTimePickerProps & WithStyles<typeof styles>) {
+    const datePickerRef = React.useRef<HTMLElement>(null);
+    const timePickerRef = React.useRef<HTMLElement>(null);
+
+    const [valueWasUndefinedBeforeLastOnChange, setValueWasUndefinedBeforeLastOnChange] = React.useState(!value);
+
+    const onChangeDateOrTime = (val?: Date | string) => {
```

Comment:
To my understanding, this only handles the undefined case. Otherwise you might expect this to be a parent function to `onChangeDate` and `onChangeTime`

```suggestion
    const onChangeDateOrTimeToUndefined = (val?: Date | string) => {
```